src/EnergyPlus/HVACFourPipeBeam.cc

```diff
@@ -0,0 +1,1302 @@
+		}
+
+	} //set_size
+
+	Real64 HVACFourPipeBeam::residualSizing(
+		Real64 const airFlow // air flow in kg/s
+	){
+
+		static std::string const routineName( "Real64 HVACFourPipeBeam::residualSizing " );
+		Real64 rho; // local fluid density
+		Real64 Residuum; // residual to be minimized to zero
+
+		this->mDotSystemAir =airFlow; 
+		this->vDotDesignPrimAir = this->mDotSystemAir / DataEnvironment::StdRhoAir ;
+
+		this->totBeamLength = this->vDotDesignPrimAir / this->vDotNormRatedPrimAir;
+		if ( this->vDotDesignCWWasAutosized ) {
+			this->vDotDesignCW = this->vDotNormRatedCW * this->totBeamLength;
+			rho = FluidProperties::GetDensityGlycol( DataPlant::PlantLoop( this->cWLocation.loopNum ).FluidName, DataGlobals::InitConvTemp, 
+									DataPlant::PlantLoop( this->cWLocation.loopNum ).FluidIndex, routineName );
+			this->mDotNormRatedCW = this->vDotNormRatedCW * rho;
+			this->mDotCW = this->vDotDesignCW * rho;
+			if ( this-> beamCoolingPresent ) {
+				PlantUtilities::InitComponentNodes(0.0, 
+										this->mDotCW, 
+										this->cWInNodeNum, 
+										this->cWOutNodeNum, 
+										this->cWLocation.loopNum, 
+										this->cWLocation.loopSideNum, 
+										this->cWLocation.branchNum, 
+										this->cWLocation.compNum 
+										);
+			}
+		}
+		if ( vDotDesignHWWasAutosized ) {
+			this->vDotDesignHW = this->vDotNormRatedHW * this->totBeamLength;
+			rho = FluidProperties::GetDensityGlycol( DataPlant::PlantLoop( this->hWLocation.loopNum ).FluidName, DataGlobals::InitConvTemp, 
+									DataPlant::PlantLoop( this->hWLocation.loopNum ).FluidIndex, routineName );
+			this->mDotNormRatedHW = this->vDotNormRatedHW * rho;
+			this->mDotHW = this->vDotDesignHW * rho;
+			if ( this-> beamHeatingPresent ) {
+				PlantUtilities::InitComponentNodes( 0.0, 
+										this->mDotHW, 
+										this->hWInNodeNum, 
+										this->hWOutNodeNum, 
+										this->hWLocation.loopNum, 
+										this->hWLocation.loopSideNum, 
+										this->hWLocation.branchNum, 
+										this->hWLocation.compNum 
+										);
+			}
+		}
+		this->calc();
+		Residuum = (  ( this->qDotZoneReq - this->qDotTotalDelivered ) 
+						/ this->qDotZoneReq );
+
+		return Residuum;
+
+	}
+
+	void
+	HVACFourPipeBeam::control(
+		bool const EP_UNUSED( FirstHVACIteration ), // TRUE if 1st HVAC simulation of system timestep
+		Real64 & NonAirSysOutput // convective cooling by the beam system [W]
+	)
+	{
+
+
+		// Using/Aliasing
+		using namespace DataZoneEnergyDemands;
+		using General::SolveRegulaFalsi;
+		using PlantUtilities::SetComponentFlowRate;
+		using namespace std::placeholders;
+
+		bool dOASMode = false ; // true if unit is operating as DOAS terminal with no heating or cooling by beam
+
+		int SolFlag;
+		Real64 ErrTolerance;
+
+		NonAirSysOutput = 0.0; // initialize
+
+		if ( this->mDotSystemAir == 0.0 || ( ! this->airAvailable  && ! this->coolingAvailable && ! this->heatingAvailable ) ) { //unit is off
```

Comment:
@EnergyArchmage This isn't specific to you but is something systemic to EnergyPlus as a whole. Doing this comparison, `mDotSystemAir == 0.0` can cause subtle issues. Floating point comparisons are finicky and hard to do correctly. We need an approach for EnergyPlus as a whole. 

This issue with that comparison is it will evaluate to false if `mDotSystemAir = 0.0000000000000001` which it could due to floating point rounding error. 

@Myoldmopar fyi